typescript/sdk/src/middleware/deploy.ts

```diff
@@ -136,47 +133,52 @@ export class InterchainAccountDeployer extends MiddlewareRouterDeployer<
     chain: ChainName,
     config: InterchainAccountConfig,
   ): Promise<InterchainAccountContracts> {
-    const proxyAdmin = await this.deployContractFromFactory(
-      chain,
-      new ProxyAdmin__factory(),
-      'proxyAdmin',
-      [],
-      { create2Salt: this.create2salt },
-    );
-
-    // 1. deploy the proxy first with a dummy implementation (ProxyAdmin)
+    const proxyAdmin = await this.deployContract(chain, 'proxyAdmin', []);
+
+    // adapted from HyperlaneDeployer.deployProxiedContract
```

Comment:
Please expand on this in a comment. Why do we need a custom implementation here?

Reply:
added color